charmhelpers/core/hookenv.py

```diff
@@ -354,21 +354,27 @@ def _implicit_save(self):
             self.save()
 
 
-@cached
+_cache_config = None
+
+
 def config(scope=None):
-    """Juju charm configuration"""
-    config_cmd_line = ['config-get']
-    if scope is not None:
-        config_cmd_line.append(scope)
-    else:
-        config_cmd_line.append('--all')
-    config_cmd_line.append('--format=json')
+    """
+    Juju charm configuration
+
+    :param scope: string. (Optional) Return value for specified key
```

Comment:
So we can do these docstring comments as:

```python
def config(scope=None):
    """
    Get the juju charm configuration (scope==None) or individual key, 
    (scope=str).  The returned value is a Python data structured loaded as 
    JSON from the Juju config command.

    :param scope: If Set, return the value for the specified key.
    :type scope: Optional[str]
    :returns: Either the whole config as a Config, or a key from it.
    :rtype: Any
    """
```

This is basically using `typing` to specify the param and return types, and Sphinx renders the docstring in a really nice way.

Reply:
This is wonderful, I was unaware of how elaborately Sphinx made use of the doc-string for rich formatting. Thank you!